x/paloma/keeper/keeper_integration_test.go

```diff
@@ -108,18 +109,21 @@ var _ = Describe("jailing validators with missing external chain infos", func()
 
 		It("jails val[1] and val[2], but no val[0]", func() {
 			By("validators are not jailed")
-			Expect(a.ValsetKeeper.IsJailed(ctx, vals[0].GetOperator())).To(BeFalse())
-			Expect(a.ValsetKeeper.IsJailed(ctx, vals[1].GetOperator())).To(BeFalse())
-			Expect(a.ValsetKeeper.IsJailed(ctx, vals[2].GetOperator())).To(BeFalse())
+			valAddress1, err := keeperutil.ValAddressFromBech32(a.PalomaKeeper.AddressCodec, vals[0].GetOperator())
+			valAddress2, err := keeperutil.ValAddressFromBech32(a.PalomaKeeper.AddressCodec, vals[1].GetOperator())
+			valAddress3, err := keeperutil.ValAddressFromBech32(a.PalomaKeeper.AddressCodec, vals[2].GetOperator())
```

Comment:
So, this back and forth of parsing formats is getting out of hand, it even transpires to test cases.

We need a better solution for this. I'm open to ideas and suggestions here. I have to ideaas that come to mind:

### a) create util functions
that are used for getting operator/validator/consesus addresses, that assume getting the  keys will usually work and look somewhat like this:

```go
func (a *Addresser) GetOperator(v sdk.ValidatorI) string {
 defer func() { if r := recover(); r != nil {
  // log error & return default value
} }()

  return address and panic if error
}
```

It's not super pretty and basically copies the approach that was now removed in Cosmos of panicking all the time.

### b) Use proper types on the functions instead of `[]byte` or `string`

Might also be worth taking a look at other chains that have upgraded to `0.50` and check how they're avoid this DRY hell.

Reply:
I applied the same util function in [keeper](https://github.com/palomachain/paloma/pull/1060/commits/729f83611bec75d58cc4bdb0770af60772e9b848#diff-19d03101c764af0a94afb12fb3fa132115792735da677d8bae5b51d001ca2f39R87)  for the above case and pushed.

Reply:
I have gone through couple of chains' OSS but none of them have migrated to 0.50 yet.

Reply:
Did you look at all of them? Some of them might have dedicated feature branches like we do.

Reply:
I did not look at each, let me traverse through all of them.

Reply:
I came across Kujira-core chain which has upgraded SDK. They have just three occurences of `StringToBytes` conversion and did not create any util file for this case. Instead they created function `ValidatorAddressCodec` within keeper.go 

```go
// ValidatorAddressCodec returns the app validator address codec.
func (k Keeper) ValidatorAddressCodec() addresscodec.Codec {
	return k.validatorAddressCodec
}
```

Reply:
So, since CosmosSDK seems to have moved away from `string` representation to `[]byte`, I suppose it makes the most sense to update consuming function signatures to use `[]byte` as well instead of `string`.

Maybe worth creating custom types like `type ValAddr []byte` which allows us to implement stringer interface on it as well, also makes the intention of the `[]byte` variable a lot more explicit.

Reply:
I guess It's not moved away, we have `bech32` `string` type in client side tools like kepler wallet  /  chain explorers, whereas `byte` type is required for core sdk backend layer.

Reply:
Looks like lots of Paloma code is working against string representation. If Cosmos has moved to `[]byte`, we should follow. Obviously we can still support `string` on the frontend.

Reply:
I pushed the modifications as you mentioned in the top of this thread.